pkg/jobs/hive/hive.go

```diff
@@ -485,6 +748,91 @@ func validateUpgradeVersion(client clientv1.Client, clusterName string, curator
 	return nil
 }
 
+func waitForMCV(client clientv1.Client, clusterName string, clusterNamespace string, mcv *managedclusterviewv1beta1.ManagedClusterView, errMsg error) error {
+	for i := 1; i <= 5; i++ {
+		time.Sleep(utils.PauseFiveSeconds)
+		if err := client.Get(context.TODO(), types.NamespacedName{
+			Name:      clusterName,
+			Namespace: clusterNamespace,
+		}, mcv); err != nil {
+			if i == 5 {
+				klog.Warning(err)
+				return errMsg
+			}
+			klog.Warning(err)
+			continue
+		}
+		labels := mcv.ObjectMeta.GetLabels()
+		if len(labels) == 0 {
+			return errMsg
+		}
+		if _, ok := labels[MCVUpgradeLabel]; ok {
+			if mcv.Status.Result.Raw != nil {
+				break
+			}
+		} else {
+			return errMsg
+		}
+	}
+
+	return nil
+}
+
+func validateEUSUpgradeVersion(client clientv1.Client, clusterName string, curator *clustercuratorv1.ClusterCurator, isInterVersion bool) error {
+	if curator.Spec.Upgrade.DesiredUpdate == "" {
+		return errors.New(fmt.Sprintf("DesiredUpdate is required to run EUS to EUS upgrade for Curator %q", curator.Name))
+	}
+	desiredVersion, err := semver.Make(curator.Spec.Upgrade.DesiredUpdate)
+	if err != nil {
+		return err
+	}
+
+	intermediateVersion, err := semver.Make(curator.Spec.Upgrade.IntermediateUpdate)
+	if err != nil {
+		return err
+	}
+
+	managedClusterInfo := managedclusterinfov1beta1.ManagedClusterInfo{}
+	if err := client.Get(context.TODO(), types.NamespacedName{
+		Namespace: clusterName,
+		Name:      clusterName,
+	}, &managedClusterInfo); err != nil {
+		return err
+	}
+
+	klog.V(2).Info("kubevendor: ", managedClusterInfo.Status.KubeVendor)
+
+	if managedClusterInfo.Status.KubeVendor != "OpenShift" {
+		return errors.New("can not upgrade non openshift cluster")
+	}
+
+	currentVersion, err := semver.Make(managedClusterInfo.Status.DistributionInfo.OCP.Version)
+	if err != nil {
+		return err
+	}
+
+	if isInterVersion && (intermediateVersion.Compare(currentVersion) == 0 || intermediateVersion.Compare(currentVersion) == -1) {
+		return errors.New(fmt.Sprintf("IntermediateUpdate %v must be greater than current version %v to run EUS to EUS upgrade for Curator %q",
+			intermediateVersion, currentVersion, curator.Name))
+	}
+
+	// desiredVersion == targeted final EUS version
+	if desiredVersion.Compare(intermediateVersion) == 0 || desiredVersion.Compare(intermediateVersion) == -1 {
+		return errors.New(fmt.Sprintf("DesiredUpdate %v must be greater than IntermediateUpdate %v to run EUS to EUS upgrade for Curator %q",
+			desiredVersion, intermediateVersion, curator.Name))
+	}
+
+	if intermediateVersion.Major != currentVersion.Major || desiredVersion.Major != currentVersion.Major {
+		return errors.New(fmt.Sprintf("Major version EUS to EUS upgrade in not supported for Curator %q", curator.Name))
+	}
+
+	if isInterVersion && (intermediateVersion.Minor != (currentVersion.Minor+1) || desiredVersion.Minor != (currentVersion.Minor+2)) {
```

Comment:
I thought Scott said something about let's relax the check a bit. I agree with the generic ones above like current < inter < final but I am not sure about this specific one. Could you double check? Thx.

Reply:
Right, Scott said we don't need to check for when users try to go to N+2. This check is just to make sure the current , inter and final are all consecutive at the minor version level. So making sure the versions are like this 4.12 -> 4.13 -> 4.14. We don't want them to do 4.12 -> 4.14 -> 4.15.